.github/workflows/dep_build_guests.yml

```diff
@@ -0,0 +1,85 @@
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+
+name: Build Guests
+
+on:
+  workflow_call:
+    inputs:
+      docs_only:
+        description: Skip building if docs only
+        required: false
+        type: string
+        default: "false"
+
+env:
+  CARGO_TERM_COLOR: always
+
+permissions:
+  id-token: write
+  contents: read
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-guests:
+    if: ${{ inputs.docs_only == 'false' }}
+    timeout-minutes: 15
+    strategy:
+      fail-fast: true
+      matrix:
+        config: [debug, release]
+    runs-on: [self-hosted, Linux, X64, "1ES.Pool=hld-kvm-amd"]
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Set up nightly rust 
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          components: rustfmt
+
+      - uses: hyperlight-dev/ci-setup-workflow@v1.8.0
+        with:
+          rust-toolchain: "1.89"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Fix cargo home permissions
+        if: runner.os == 'Linux'
+        run: |
+          sudo chown -R $(id -u):$(id -g) /opt/cargo || true
+          sudo chmod -R u+rwX /opt/cargo || true
+
+      - name: Rust cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          shared-key: "guests-${{ inputs.config }}"
+          cache-on-failure: "true"
```

Comment:
this does not restore guest cache currently, probably need some more tuning, see https://github.com/hyperlight-dev/hyperlight/actions/runs/20116574753/job/57731955963#step:6:32